### PR TITLE
Add assertion message to assert statement in ModuleHook

### LINKED ISSUE #4626: PyInstaller 3.6 AssertionError
Hi, since the latest 3.6 release all my Linux builds are failing with a new `AssertionError`.
This is the log:
```
23 INFO: PyInstaller: 3.6
24 INFO: Python: 2.7.16
25 INFO: Platform: Linux-4.9.184-linuxkit-x86_64-with-debian-squeeze-sid
26 INFO: wrote /simulator.spec
35 INFO: UPX is not available.
36 INFO: Removing temporary files and cleaning cache in /.cache/pyinstaller
37 INFO: Extending PYTHONPATH with paths
['/agent/project/src/bin', '/']
37 INFO: checking Analysis
37 INFO: Building Analysis because Analysis-00.toc is non existent
38 INFO: Initializing module dependency graph...
49 INFO: Caching module graph hooks...
Traceback (most recent call last):
  File "/usr/local/bin/pyinstaller", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/__main__.py", line 114, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/building/build_main.py", line 734, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/building/build_main.py", line 681, in build
    exec(code, spec_namespace)
  File "/simulator.spec", line 17, in <module>
    noarchive=False)
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/building/build_main.py", line 244, in __init__
    self.__postinit__()
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/building/datastruct.py", line 160, in __postinit__
    self.assemble()
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/building/build_main.py", line 343, in assemble
    excludes=self.excludes, user_hook_dirs=self.hookspath)
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/analysis.py", line 734, in initialize_modgraph
    user_hook_dirs=user_hook_dirs,
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/analysis.py", line 116, in __init__
    self._reset(user_hook_dirs)
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/analysis.py", line 134, in _reset
    self._hooks = self._cache_hooks("")
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/analysis.py", line 215, in _cache_hooks
    return ModuleHookCache(self, hook_dirs)
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/imphook.py", line 100, in __init__
    self._cache_hook_dirs(hook_dirs)
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/imphook.py", line 133, in _cache_hook_dirs
    hook_module_name_prefix=self._hook_module_name_prefix,
  File "/usr/local/lib/python2.7/site-packages/PyInstaller/depend/imphook.py", line 289, in __init__
    assert self.hook_module_name not in HOOKS_MODULE_NAMES
AssertionError
```

PyInstaller: 3.6
OS: Linux & macOS
Python: 2.7

After reverting to version 3.5 it seems to work fine, what can I do to try to find the source of this issue?

### PR DESCRIPTION
**Pull request by Legorooj**
Closes #4626 
Related to #4538 
Supercedes #4684

### CODE REVIEW

**Comment by Legorooj** [comment]
@bjones1 anything else to do?

File: PyInstaller/depend/imphook.py [hunk 1]
```diff
@@ -285,7 +285,9 @@ def __init__(self, module_graph, module_name, hook_filename,
 
         # Safety check, see above
         global HOOKS_MODULE_NAMES
-        assert self.hook_module_name not in HOOKS_MODULE_NAMES
+        assert self.hook_module_name not in HOOKS_MODULE_NAMES, \
+            'Hook Conflict. Please remove the following custom hook: ' \
+            '{hook}'.format(hook=self._hook_name)
```
Comment (bjones1) [inline PyInstaller/depend/imphook.py]: A quick question -- it `self._hook_name` correct? I can't find this being set anywhere in the code. Have you tested that this produces the correct error message?

**Comment by logan-pugh** [comment]
How does this change impact https://github.com/pyinstaller/pyinstaller/issues/1835?

What will be the mechanism for overriding the behavior of default hooks?

File: PyInstaller/depend/imphook.py (hunk 1 above)
Comment (Legorooj) [reply PyInstaller/depend/imphook.py]: @bjones1 Umm? I meant to type `self.hook_module_name`. Sometimes autocomplete isn't helpful.

**Comment by Legorooj** [comment]
@logan-pugh it doesn't. The best solution for now is to edit the matplotlib hook file (`PyInstaller.hook.hook-matplotlib`) with an if statement that checks for an environment variable (`COPY_MPL_DATA`), and if that is true, you copy the data, if not, etc.

**Comment by Legorooj** [comment]
@bjones1

**Comment by bjones1** [comment]
I notice that one of the commits doesn't follow the comment message conventions. Since this is a small change, would you squash all these commits into one and title that commit per the conventions?

**Comment by Legorooj** [comment]
@bjones1 to save time can't you squash merge? Or has harmut disabled that?

**Comment by bjones1** [comment]
Ah, I didn't see that. Thanks for pointing it out.

**Pull request merged**